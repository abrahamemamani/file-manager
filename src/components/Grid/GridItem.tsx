import React, { FC, ReactNode } from "react";

export interface GridItemProps {
  children: ReactNode;
  className?: string | undefined;
}

export const GridItem: FC<GridItemProps> = ({ children, className = "" }) => (
  <div className={`${className}`}>{children}</div>
);
