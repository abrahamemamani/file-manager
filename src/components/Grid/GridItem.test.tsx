import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { GridItem, GridItemProps } from "./GridItem";

const defaultProps: GridItemProps = {
  children: <span>Children</span>,
};

describe("[components/Grid]", () => {
  describe("when GridItem is rendered", () => {
    it("should render", () => {
      const { asFragment } = render(<GridItem {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
