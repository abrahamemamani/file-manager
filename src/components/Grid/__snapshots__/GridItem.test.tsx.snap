// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[components/Grid] > when GridItem is rendered > should render 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <span>
      Children
    </span>
  </div>
</DocumentFragment>
`;
